worker/workercmd/worker_linux.go

```diff
@@ -22,26 +22,39 @@ type Certs struct {
 	Dir string `long:"certs-dir" description:"Directory to use when creating the resource certificates volume."`
 }
 
-type GardenBackend struct {
-	UseHoudini    bool `long:"use-houdini"    description:"Use the insecure Houdini Garden backend."`
-	UseContainerd bool `long:"use-containerd" description:"Use the containerd backend. (experimental)"`
+type Runtime struct {
+	Type string `long:"type" default:"guardian" description:"Runtime to use with the worker. Possible values: guardian, containerd, houdini. Please note that Houdini is insecure and doesn't run 'tasks' in containers."`
```

Comment:
`--runtime=containerd` feels better than `--runtime-type=containerd`, WDYT?

Reply:
This choice came down to thinking about runtimes available for Linux vs non-Linux platforms. "Runtime" is defined at the `worker.go` level. If we want different behaviours for `worker_linux.go` and `worker_nonlinux.go` we had to make it a struct that could be defined differently in those two files. This mean we can easily enable only Houdini in non-Linux, and not have to write out our own validation.

Although I suppose the validation wouldn't may not be too difficult to do in each of those files simply based off the `cmd.Runtime` value. @xtreme-sameer-vohra any thoughts on this?

Reply:
@chenbh the other issue here was the message we show for `Runtime` has to be different for linux vs non-linux.

e.g.
```
type Runtime struct {
	Type string `long:"type" default:"guardian" description:"Runtime to use with the worker. Possible values: guardian, containerd, houdini. Please note that Houdini is insecure and doesn't run 'tasks' in containers."`
}
```
for Linux
versus
```
type Runtime struct {
}
```
providing no option at all for non-linux as it defaults to Houdini. We need this structure as in the future we will likely see containerd enabled for Windows, and then have to introduce type here and a message saying only `containerd` and `houdini` are allowed values. (We will also likely have to break that out into a worker_mac and worker_windows file at that point with the worker_mac still not allowing a value and defaulting to houdini.)

Reply:
Should be possible with struct inheritance

worker.go:
```golang
type WorkerCommand struct {
...
	Runtime `group:"Runtime Configuration"`
...
}
```

worker_linux.go:
```golang
// +build linux
type Runtime struct {
	Runtime string `long:"runtime" default:"guardian" choice:"guardian" choice:"containerd" choice:"houdini" description:"Runtime to use with the worker. Please note that Houdini is insecure and doesn't run 'tasks' in containers."`
}
```

worker_nonlinux.go:
```golang
// +build !linux
type Runtime struct {}
```

Reply:
@chenbh Amazing! We tried that initially but didn't realize you could set the parent struct and not have to provide the `namespace` attribute. Made the suggested change to `CONCOURSE_RUNTIME` in BOSH release and Helm chart as well. Thanks!